clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1675,6 +1677,20 @@ public static <V> Map<String, V> entriesWithPrefix(Map<String, V> map, String pr
         return result;
     }
 
+    /**
+     * Sanitize the string by limiting the max length and trim the spaces (i.e. \s\r\n\t)
+     * @param str the original string
+     * @param len the max length of the string
+     * @return sanitized string
+     */
+    public static String sanitizeString(String str, int len) {
```

Comment:
i'm wondering whether we can have
```
    public static String sanitizeString(String str) {
        return sanitizeString(str, USERNAME_MAX_LEN);
    }
```
and use this instead of passing USERNAME_MAX_LEN across all invocations